pkg/model/job_selection.go

```diff
@@ -46,3 +46,13 @@ func NewDefaultJobSelectionPolicy() JobSelectionPolicy {
 		Locality: Anywhere,
 	}
 }
+
+func (e JobSelectionDataLocality) MarshalText() ([]byte, error) {
+	return []byte(e.String()), nil
+}
+
+func (e *JobSelectionDataLocality) UnmarshalText(text []byte) (err error) {
+	name := string(text)
+	*e, err = ParseJobSelectionDataLocality(name)
+	return
+}
```

Comment:
This is dead code: I was attempting to allow `BacalhauConfig` to contain the actual values we use in the code, but for the values to be read and written from the config as human readable values (e.g. `JobSelectionDataLocality` in the config could sting ("local"|"anywhere", but read as a type (1, 2), I couldn't get it to work)

Reply:
That would be neat if we can get an extra level of type safety, from what I can tell with a cursory glance it seems like a lot of boiler plate to do this with viper/mapstructure/decodehooks - hopefully you can find a way.

Reply:
Brilliant, just needed to register the text decoder, custom types work!
```golang
	if err := viper.Unmarshal(&cfg, viper.DecodeHook(mapstructure.TextUnmarshallerHookFunc())); err != nil {
		return err
	}
```